builder/docker/builder.go

```diff
@@ -23,15 +23,14 @@ func NewImageBuilder(executor exec.Executor, dryRun bool) *ImageBuilder {
 // Build the image using the docker executable.
 // If the image is built successfully, the image will be pushed to the registry.
 func (b ImageBuilder) Build(opts types.ImageBuilderOpts) error {
-	if b.dryRun {
-		logrus.Infof("[DRY-RUN] docker build --no-cache --pull -t %s %s", opts.Tag, opts.Context)
+	if b.dryRun && !opts.LocalOnly {
+		logrus.Infof("[DRY-RUN] docker build --no-cache -t %s %s", opts.Tag, opts.Context)
 		logrus.Infof("[DRY-RUN] docker push %s", opts.Tag)
 		return nil
 	}
```

Comment:
Attention, si `opts.LocalOnly` est `true`, le mode dry run sera toujours désactivé
Ce serait plutôt :
```suggestion
	if b.dryRun {
		logrus.Infof("[DRY-RUN] docker build --no-cache -t %s %s", opts.Tag, opts.Context)
		if !opts.LocalOnly {
			logrus.Infof("[DRY-RUN] docker push %s", opts.Tag)
		}
		return nil
	}
```